File: src/main/java/ca/weblite/jdeploy/app/services/NpmAccountService.java
package ca.weblite.jdeploy.app.services;

import ca.weblite.jdeploy.app.exceptions.NotFoundException;
import ca.weblite.jdeploy.app.records.NpmAccount;
import ca.weblite.jdeploy.app.repositories.NpmAccountRepository;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.UUID;

@Singleton
public class NpmAccountService {
    private final NpmAccountRepository npmAccountRepository;

    @Inject
    public NpmAccountService(NpmAccountRepository npmAccountRepository) {
        this.npmAccountRepository = npmAccountRepository;
    }

    public NpmAccount findOneById(UUID uuid) throws NotFoundException {
        return npmAccountRepository.findOneById(uuid);
    }

    public NpmAccount findOneByIdOrNull(UUID uuid) {
        return npmAccountRepository.findOneByIdOrNull(uuid);
    }
}

File: src/main/java/ca/weblite/jdeploy/app/services/NpmAccountService.kt
package ca.weblite.jdeploy.app.services

import ca.weblite.jdeploy.app.exceptions.NotFoundException
import ca.weblite.jdeploy.app.records.NpmAccount
import ca.weblite.jdeploy.app.repositories.NpmAccountRepositoryInterface
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class NpmAccountService @Inject constructor (
    private val npmAccountRepository: NpmAccountRepositoryInterface
) {
    @Throws(NotFoundException::class)
    fun findOneById(uuid: UUID): NpmAccount {
        return npmAccountRepository.findOneById(uuid)
    }

    fun saveOne(npmAccount: NpmAccount): NpmAccount {
        return npmAccountRepository.saveOne(npmAccount)
    }
}
